utils/python/CIME/SystemTests/system_tests_common.py

```diff
@@ -49,17 +49,28 @@ def build(self, sharedlib_only=False, model_only=False):
                          sharedlib_only=sharedlib_only, model_only=model_only)
 
     def run(self):
+        return self._run()
+
+    def _run(self, suffix="base"):
         rc, out, err = run_cmd("./case.run --caseroot %s" % self._caseroot, ok_to_fail=True)
-        if rc == 0 and self.coupler_log_indicates_run_complete():
-            self._runstatus = "PASS"
+        if rc == 0 and self._coupler_log_indicates_run_complete():
+            success = True
+            if self._runstatus != "FAIL":
+                self._runstatus = "PASS"
         else:
+            success = False
             self._runstatus = "FAIL"
 
         if out:
             append_status("case.run output is:\n %s\n"%out, sfile="TestStatus.log")
         if err:
             append_status("case.run error is:\n %s\n"%err, sfile="TestStatus.log")
 
+        if success and suffix is not None:
```

Comment:
Successful runs often need to change the name of the hist files to something so subsequent runs can be compared.